File: testdata/go/expected.go
//go:build linux && !windows
// +build linux,!windows

/*
Package documentation

	spanning multiple lines
*/
package main

import (
	"fmt"
)

const Version = "v1.0.0"

/* Block comment before var */
var name = /* mid-token */ "Gopher"
var age = 10 /* trailing block */

func hello() {
	fmt.Println("Hello") /* inline block comment */
}

/*
 * Multi-line block
 * comment inside file
 */
func main() {
	hello()

	//go:generate echo "generate something"
	//go:noinline

	if true {
		fmt.Println("Conditional")
	} else /* else comment */ {
		fmt.Println("Else branch")
	}

}

File: testdata/kotlin/expected.kt
package com.example

import kotlin.collections.*

/**
 * Class documentation comment
 * @property name The name property
 */
@Entity
@Suppress("UNCHECKED_CAST")
class Person(
    private val name: String,
    val age: Int
) {
    private val id: Long = 0L
    
    /**
     * Function documentation 
     * @param greeting The greeting message
     * @return formatted greeting
     */
    @JvmStatic
    fun greet(greeting: String): String {
        return "$greeting, $name!"
    }
    
    /* 
     * Multi-line comment block
     * with multiple lines
     */
    private fun validate() {
        if (name.isEmpty()) {
            throw IllegalArgumentException("Name cannot be empty")
        }
    }
}

@Deprecated("Use factory method instead")
object PersonFactory {
    fun createPerson(name: String, age: Int): Person {
        /* Block comment before return */
        return Person(name, age)
    }
} 